### PR TITLE
feat(eth payments): PE-5992

### PR DESCRIPTION
**Pull request by fedellen**
This PR adds support for `ethereum` token in the `topUpWithTokens` method

### CODE REVIEW

File: src/common/signer.ts [hunk 1]
```diff
@@ -16,10 +16,13 @@
  */
 import { EthereumSigner, HexSolanaSigner } from 'arbundles';
 import { randomBytes } from 'crypto';
+import 'ethers';
```
Comment (fedellen) [inline src/common/signer.ts:19]:
```suggestion
```

File: src/common/ethereum.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import {
+  TokenConfig,
+  TokenCreateTxParams,
+  TokenPollingOptions,
+  TokenTools,
+  TurboLogger,
+} from '../types.js';
+import { TurboWinstonLogger } from './logger.js';
+
+export class EthereumToken implements TokenTools {
```
Comment (fedellen) [inline src/common/ethereum.ts:28]: struggling to get test coverage via a testnet/localnet

Comment (fedellen) [reply src/common/ethereum.ts:28]: Managed to find an ETH testnet that worked for some coverage

**Review by dtfiedler — APPROVED** [review]

**Pull request merged**